crates/claims/crates/status/src/client/mod.rs

```diff
@@ -0,0 +1,78 @@
+use crate::EncodedStatusMap;
+use std::{borrow::Borrow, ops::Deref, sync::Arc};
+
+mod cache;
+pub use cache::Cached;
+
+mod http;
+pub use http::HttpClient;
+
+#[derive(Debug, thiserror::Error)]
+pub enum ProviderError {
+    #[error("not found")]
+    NotFound,
+
+    #[error("internal error: {0}")]
+    Internal(String),
+
+    #[error("missing media type")]
+    MissingMediaType,
+
+    #[error("invalid media type")]
+    InvalidMediaType,
+
+    #[error("{0}")]
+    Encoded(String),
+
+    #[error("decoding failed: {0}")]
+    Decoding(String),
+}
+
+pub trait TypedStatusMapProvider<I: ?Sized, T: EncodedStatusMap> {
+    #[allow(async_fn_in_trait)]
```

Comment:
This is safe because `id` isn't Sized?

Reply:
Its always safe. Are you talking about the `allow`? It's just because the compiler like to warn us that users can't put bounds on the return type of those functions. But we don't care here.